fhd_core/setup_metadata/fhd_struct_init_obs.pro

```diff
@@ -132,7 +128,21 @@ IF N_Elements(min_baseline) EQ 0 THEN min_baseline=Min(kr_arr[where(kr_arr)]) EL
 kx_arr=0 & ky_arr=0 & kr_arr=0 ;free memory
 noise_arr=Ptr_new()
 
-meta=fhd_struct_init_meta(file_path_vis,hdr,params,degpix=degpix,dimension=dimension,elements=elements,$
+; If it appears that the params struct uses tile names instead of indices, then rewrite with indices
+if max(params.antenna1) GT n_tile then begin
+    tile_A = params.antenna1
+    tile_B = params.antenna2
+    for tile_i=0, n_tile-1 do begin
+        inds = where(layout.antenna_numbers[tile_i] EQ (params.antenna1),n_count)
+        if n_count GT 0 then tile_A[inds] = tile_i+1
+        inds = where(layout.antenna_numbers[tile_i] EQ (params.antenna2),n_count)
+        if n_count GT 0 then tile_B[inds] = tile_i+1
```

Comment:
Should you set `tile_A` and `tile_B` differently if `n_count` is ever 0? I could imagine a situation where a tile was flagged and possibly not in `layout.antenna_numbers`, and it could either have the same value of `params.antenna1` (or `2`) as an existing tile index, or it could be greater than `n_tile`. Either way could lead to errors. One option could also be to initialize `tile_A` and `tile_B` to something like -1, and print a warning if there are any -1's remaining after this tile number reassignment.

Reply:
If `n_count` is zero, then the submitted uvfits file is broken. `params.antenna1/2` are set from the uvfits visibility extension baseline or antenna arrays. These arrays can only have entries which are also in the antenna extension antenna array, which is what `layout.antenna_numbers` comes from. So maybe we should throw an error if `n_count` is zero?

Reply:
Yes, good idea to throw an error if that means the file is broken.

Reply:
actually I have that wrong. This is iterating over `layout.antenna_numbers`; it is possible for an antenna to be present in `layout.antenna_numbers` and not be present in `params.antenna1/2`. So if `n_count` is zero, that antenna is not present in `params.antenna1/2` and thus in not present in `tile_A/B`. No extra handling is necessary if `n_count` is zero.